### PR TITLE
Issue #4727 setConfigurationDiscovered true per default

### PR DESCRIPTION
**Pull request by olamy**

### CODE REVIEW

**Review by janbartel — CHANGES_REQUESTED** [review]
The jetty documentation on Annotations needs to be updated to include the Configuration.setConfigurationDiscovered(boolean) method, and the fact that it will default to TRUE for even 2.5 webapps.

File: jetty-documentation/src/main/asciidoc/distribution-guide/annotations/using-annotations.adoc [hunk 1]
```diff
@@ -121,9 +121,8 @@ As is the case with annotation scanning, the link:#using-extra-classpath-method[
 
 ____
 [NOTE]
-As of Jetty-9.4.4, unless the `web.xml` is version 3.0 or greater, only `ServletContainerInitializers` that are on the container classpath will be discovered.
-Users wishing to use `ServletContainerInitializers` from within the webapp with older versions of `web.xml` must either upgrade their `web.xml` version, or call `WebAppContext.setConfigurationDiscovered(true)` either programmatically or in xml.
-Upgrading the `web.xml` version is preferable.
+As of Jetty 10, Annotations will be discovered even for old versions of `web.xml` (2.5).
+Users wishing to not use `ServletContainerInitializers` from within the webapp with older versions of `web.xml` must call `WebAppContext.setConfigurationDiscovered(false)` either programmatically or in xml.
```
Comment (janbartel) [inline jetty-documentation/src/main/asciidoc/distribution-guide/annotations/using-annotations.adoc]: Actually it's not just SCIs, it is also @WebServlet, @WebFilter, @WebListener @MultiPart etc etc. So maybe phrase this as:
"Users wishing not to use annotations from the webapp classpath must call ...."

**Review by janbartel — APPROVED** [review]
LGTM with a slight wording change.

**Review by joakime — APPROVED** [review]
LGTM, just need to work out why the build failed.

**Pull request merged**